### ISSUE TITLE
week.php XSS init.php

### ISSUE THREAD

**Issue by hyperbolicTom** [craigk5n/webcalendar#260]
https://hostname/cal/week.php/sssssssss%22onload=%22alert(document.cookie)

"The page shows the url path in the body. This is not filtered/checked/escaped. This allows me or an attacker to
inject out own code into the page, and get an xss."

This puts
`  <body id="sssssssss"onload="alert(document.cookie)">`
into the page.

So this code in print_header():

```
  /* Add <body> id. */ . ' id="' . preg_replace ( '/(_|.php)/', '',
    substr ( $self, strrpos ( $self, '/' ) + 1 ) ) . '"'

```
Can somebody suggest a good way to fix this? $self needs to be sanitized, somwhere.

our pen-tester is Synack btw

**Comment by hyperbolicTom** [comment]
I will try this:

`  $selfA = explode("/", $self);

  $self = $selfA[0] . "/" . $selfA[1] . "/" . $selfA[2];
`

**Comment by craigk5n** [comment]
Fix in f1c147b6b122393f8adeb6fba4b69188ec33d647 now in bootstrap-ui branch.  Will be included in next release.
